tutorials/math/random_number_generation.rst

```diff
@@ -0,0 +1,295 @@
+.. _doc_random_number_generation:
+
+Random number generation
+========================
+
+Many games rely on randomness to implement core game mechanics. This tutorial
+guides you through common types of randomness and how to implement them in
+Godot.
+
+.. note::
+
+    Computers cannot generate "true" random numbers. Instead, they rely on
+    *psuedorandom number generators* (PRNGs).
+
+Global scope versus RandomNumberGenerator class
+-----------------------------------------------
+
+Godot exposes two ways to generate random numbers: via *global scope* methods
+or using the :ref:`class_RandomNumberGenerator` class.
+
+Global scope methods are easier to set up, but they don't offer as much control.
+
+RandomNumberGenerator requires more code to use, but exposes many methods not
+found in global scope such as
+:ref:`randi_range() <class_RandomNumberGenerator_method_randi_range>` and
+:ref:`randfn() <class_RandomNumberGenerator_method_randfn>`. On top of that,
+it allows creating multiple instances each with their own seed.
+
+This tutorial uses global scope methods, except when the method is only found in
+the RandomNumberGenerator class.
+
+The randomize() method
+----------------------
+
+In global scope, you can find a :ref:`randomize() <class_@GDScript_method_randomize>`
+method.
+**This method should be called only once when your project starts to initialize
+the random seed.** Calling it multiple times is unnecessary and may impact
+performance negatively.
+
+Putting it in your main scene script's ``_ready()`` method is a good choice::
+
+    func _ready():
+        randomize()
+
+You can also set a fixed random seed instead using
+:ref:`seed() <class_@GDScript_method_seed>`. This will give you *deterministic*
+results across runs::
+
+    func _ready():
+        seed(12345)
+        # To use a string as a seed, you can hash it to a number.
+        seed("Hello world".hash())
+
+When using the RandomNumberGenerator class, you should call ``randomize()``
+on the instance since it has its own seed::
+
+    var rng = RandomNumberGenerator.new()
+    rng.randomize()
+
+Get a random number
```

Comment:
In part, this section paraphrases or duplicates the class reference's content. I'd invite people to go read the page. While I understand it's nice to have all the explanations on one page, in a reference manual, I'd try to push people to read the code reference.

It's not that I find this section bad; instead, I feel that's the perfect amount of info and detail we should have in the class ref itself.

Reply:
I wrote that section mainly to have an "overview" of all methods you can use. In hindsight, it's true we could move these examples to the class reference, but the class reference isn't always easy to browse when you're looking for various unrelated methods.